lib/dns.js

```diff
@@ -189,6 +189,9 @@ exports.lookupService = function(host, port, callback) {
   if (cares.isIP(host) === 0)
     throw new TypeError('"host" argument needs to be a valid IP address');
 
+  if (typeof port !== 'number')
+    throw new TypeError('"port" argument must be a number');
```

Comment:
Might be a good idea to add the failing value here: 

    throw new TypeError(`"port" argument must be a number, got ${port} instead.`);

That's not a regression from how other `TypeError`s are thrown here though so not a biggie.